service/src/LevelZeroDevicePool.cpp

```diff
@@ -164,6 +165,35 @@ namespace geopm
                                          dev_subdev_idx_pair.second);
     }
 
+    double LevelZeroDevicePoolImp::frequency_step(int domain, unsigned int domain_idx,
+                                                  int l0_domain) const
+    {
+        double frequency_step_mhz = NAN;
+        if (domain != GEOPM_DOMAIN_GPU_CHIP) {
+            throw Exception("LevelZeroDevicePool::" + std::string(__func__) +
+                             ": domain " + std::to_string(domain) +
+                            " is not supported for the frequency domain.",
+                            GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+        std::pair<unsigned int, unsigned int> dev_subdev_idx_pair;
+        dev_subdev_idx_pair = subdevice_device_conversion(domain_idx);
+        check_domain_exists(m_levelzero.frequency_domain_count(dev_subdev_idx_pair.first,
+                                                               l0_domain), __func__,
+                                                               __LINE__);
+
+        std::vector<double> supported_frequency = m_levelzero.frequency_supported(dev_subdev_idx_pair.first,
+                                                                                  l0_domain,
+                                                                                  dev_subdev_idx_pair.second);
+        if (supported_frequency.size() >= 2) {
+            std::sort(supported_frequency.begin(), supported_frequency.end());
+            frequency_step_mhz = (double) (supported_frequency.back() -
+                                           supported_frequency.front()) /
+                                          (supported_frequency.size() - 1);
```

Comment:
Is this expected to resolve to an integer in the 100s of MHz?

Reply:
From sysman we're getting an "array of frequencies in units of MHz."